compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -456,12 +483,13 @@ class TreePickler(pickler: TastyPickler) {
             args.foreach(pickleTpt)
           }
         case Literal(const1) =>
-          pickleConstant {
-            tree.tpe match {
-              case ConstantType(const2) => const2
-              case _ => const1
+          if assertForBestEffort(tree.hasType) then
```

Comment:
Similar thing here. There is no reason not to `pickleConstant(const1)` if the tree does not have a type.